src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -296,13 +299,17 @@ private static T NextOrDefault<T>(IEnumerator<T> enumerator)
             /// <summary>
             /// Return all the spans that appear in only one of "latestSpans" or "previousSpans".
             /// </summary>
-            private static IEnumerable<SnapshotSpan> Difference<T>(IEnumerable<ITagSpan<T>> latestSpans, IEnumerable<ITagSpan<T>> previousSpans, IEqualityComparer<T> comparer)
+            private static DiffResult Difference<T>(IEnumerable<ITagSpan<T>> latestSpans, IEnumerable<ITagSpan<T>> previousSpans, IEqualityComparer<T> comparer)
                 where T : ITag
             {
-                var latestEnumerator = latestSpans.GetEnumerator();
-                var previousEnumerator = previousSpans.GetEnumerator();
-                try
+                using (var addedPool = SharedPools.Default<List<SnapshotSpan>>().GetPooledObject())
```

Comment:
Does this intelligently free the list?  If not, i'm sure there are methods for this.  like AllocateAndFree and whatnot.